sort/mergesort.go

```diff
@@ -60,3 +60,29 @@ func MergeIter[T constraints.Ordered](items []T) []T {
 	}
 	return items
 }
+
+// MergeParallel Perform merge sort on a slice using goroutines
+func MergeParallel[T constraints.Ordered](items []T) []T {
+
+	if len(items) < 2 {
+		return items
+
+	}
+
+	if len(items) < 2048 {
+		return Merge(items)
+	}
+
+	done := make(chan struct{})
+	defer close(done)
+
+	var middle = len(items) / 2
+	var a []T
+	go func() {
+		a = MergeParallel(items[:middle])
+		done <- struct{}{}
+	}()
+	var b = MergeParallel(items[middle:])
+	<-done
```

Comment:
non-blocking but I think wait groups might be better here (which internally use this anyways but they have more protection built into them)